Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -31,10 +31,10 @@ public SystemObserver(Context context) {
 		isRealHardwareId = true;
 	}
 	
-	public String getUniqueID() {
+	public String getUniqueID(boolean debug) {
 		if (context_ != null) { 
 			String androidID = Secure.getString(context_.getContentResolver(), Secure.ANDROID_ID);
-			if (androidID == null) {
+			if (androidID == null || debug) {
```

Comment:
So if debug == true, androidID will be a random UUID anyway, why don't we do this:
String androidID = null;
if (!debug) {
    androidID = Secure.getString(context_.getContentResolver(), Secure.ANDROID_ID);
}

if (androidID == null) {
    androidID = UUID.randomUUID().toString();
    isRealHardwareId = false;
}